Ci and restructure

Pi3dra opened the pull request:
Added CI to:
- Lint decret.py and run pytests
- Build and push some example containers to GHCR
- Run trivy scans on these containers 

Also re-organized the code into different folders like decret/ and tests/,
And updated dome dependencies

Commit:
Adding flag for easier CI/CD integration

Pi3dra committed:
Update .gitignore.

Pi3dra committed:
Restructure the repo for qol.

Pi3dra committed:
First CI script (pylint + pytest).

Pi3dra committed:
Add CI pipeline for decret.py

pictyeye commented:
Thank you Pedro for your work adding a CI pipeline to the project. I believe this should be merged. @dejonnic can you accept the PR?

dejonnic commented:
Hello @Pi3dra, thank you for your valuable contributions!

Here are some suggestions for improvement:

- In your workflow file, there are some static references to your account name when pushing to GHCR. It would be beneficial to abstract these references and add some instructions/documentation to set up the workflow for other developers (steps and required environment variables).
- Given that we currently have only a single main file (decret.py), for simplicity, it might be better to keep it at the project root for now. However, as the project grows, it would be advisable to organize files into appropriate directories.

Pi3dra committed:
Abstracting static references

Also removed old gitlab CI

Pi3dra committed:
Adding small readme

Pi3dra commented:
Thanks @dejonnic for the feedback!
- I’ve abstracted the account name used to push to GHCR by referencing ${{ github.repository_owner }} instead of my static username.
- I also added a short README explaining what the workflows do and how to interpret the results from Trivy.

Also there's no particular manual configuration for the workflow to work, other than adding the branch name in the workflow.

Regarding the project structure, I understand that reorganizing it now might seem unnecessary. However, I'm currently working on improving decret, and this re-structure becomes useful later on.

dejonnic commented:
Thank you @Pi3dra for the changes.

- The Trivy workflow fails, could you please review it for potential typos or configuration issues?
- I understand you would like to change the structure because of the upcoming improvements you mentioned. In such a case,  please update the main README file and any relevant examples. This will help new contributors easily understand the project setup and get started quickly.
- There are additional packages in ' requirements.txt' that do not seem necessary for the project as is, please consider removing them for the sake of atomicity.

Thank you once again for your inputs.

Pi3dra commented:
Thanks again @dejonnic  for the feedback, and sorry for the hiccups.
I'll update the requirements, README, and examples as requested.
That said, I'm not seeing any errors with the Trivy workflow on my end ,could you please share the error message you're encountering?

Pi3dra committed:
Updating dependencies

- Updated to newer versions
- Removed pip-audit and it's dependencies
- Removed Bs4 and Pandas which are not used yet

Pi3dra committed:
Adding __main__.py for ease of use

And some quick fixes to the CI

Pi3dra committed:
Updating Examples and README

dejonnic commented:
Thank you @Pi3dra, we are converging for a merge soon. The last issues are:

- The Trivy workflow fails because of a typo, e.g. l. 37 of file "trivy-docker-test.yml" "vers" instead of "vars" of what I can see
- Now the build workflow fails as well, because of the last introduced changes in the module. I think it's enough to set "python3 -m decret" l. 78 of file "build-and-push.yml"

Thanks!

Pi3dra commented:
@dejonnic I've fixed and tested everything, it should be good to go now.
I've also dropped and squashed some commits for a cleaner history. Let me know if you'd like any further cleanup. Thanks!

dejonnic commented:
Thanks @Pi3dra, looks very good. Thank you for your efforts and this PR! Merging with main.
